src/commands/roll.ts

```diff
@@ -1,119 +1,196 @@
-                return;
-            } else {
-                embed = new EmbedBuilder()
-                    .setTitle('YOU PULLED A LEGENDARY JEFFREY!!!')
-                    .setAuthor({ name: `${interaction.user.displayName}` })
-                    .setImage(gacha)
-                    .addFields({ name: `${legendaryInfo[0]}`, value: `${legendaryInfo[1]}` });
+            let starArray = '';
+            for (let j = 0; j < gachaLevel[i]; j++) {
+                starArray += '⭐';
             }
+            let lvlUp = 'Level: ';
+            if (gachaLevel[i] > 1) {
+                lvlUp = '**Rank Up!** | Level: ';
+            }
+            level.push(lvlUp + starArray);
+
+            const color = await getEmbedColor(gacha[i].rarity);
+
+            const embed = new EmbedBuilder();
+            embed.setTitle(`${gacha[i].name} Jeffrey`)
+                .setDescription(`${gacha[i].description}`)
+                .setImage(gacha[i].link)
+                .setFields({ name: ' ', value: `${level[i]}` })
+                .setColor(color)
+                .setFooter({ text: `${0 + i + 1}/${rolls}` });
+
+            embeds.push(embed);
         }
 
-        const gachaLevel = await checkGachaLevel(userID, gacha);
-        let starArray = '';
-        if (!gachaLevel) {
-            console.log(`gachaLevel is ${gachaLevel} (NULL)`);
-            await interaction.reply('Sorry the command could not be completed, please contact a developer.')
+        if (rolls === 1) {
+            b.next.setDisabled(true);
         }
-        for (let i = 0; i < gachaLevel; i++) {
-            starArray += '⭐';
+        //previous starts disabled
+        const row = new ActionRowBuilder<ButtonBuilder>()
+            .setComponents(b.previous, b.next, b.rollAgain);
+
+        let currentGacha = 0;
+
+        if (rollingAgain && rollingAgain > 0) {
+            await interaction.channel?.send(`${interaction.user} rolled for ${rolls} Jeffrey(s)!`);
+        } else {
+            await interaction.reply(`${interaction.user} rolled for ${rolls} Jeffrey(s)!`);
         }
-        embed.setDescription(starArray);
-        try {
-            await replyWithEmbed(embed!, interaction);
-        } catch {
-            console.log(`ERROR: could not reply with embed in ${interaction.channelId}`);
+        const gachaMessage = await interaction.channel?.send({
+            content: `**${gacha[currentGacha].rarity.toUpperCase()} JEFFREY**`,
+            embeds: [embeds[currentGacha]],
+            components: [row]
+        });
+
+        if (!gachaMessage) {
+            await interaction.reply('Could not send message embed, please contact a developer');
             return;
         }
+
+        const buttonCollector = gachaMessage.createMessageComponentCollector({ filter: i => i.user.id === interaction.user.id, componentType: ComponentType.Button, time: 1_800_000 });
+
+        buttonCollector.on('collect', async i => {
+
+            if (i.customId === 'next') {
```

Comment:
Don't hardcode IDs here, just use the custom ID that's defined in the button instance (something like `BUTTONS.next.customId`)